Editing - Secure and Access Elastic Stack services (#3729)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/accessing-elastic-services.asciidoc

[id="{p}-{page_id}"]
= Access Elastic Stack services

To access the Elastic Stack services, you need to:

- Retrieve the `elastic` user password for basic authentication.
- Specify the IP of the service, if you want to access the service from outside the Kubernetes cluster.

> - Decide if you want to use the self-signed certificate with the custom CA (Certificate Authority) generated by ECK, or configure your own certificate.

Reviewer: ```suggestion
- Decide if you want to use the self-signed certificate generated by ECK, or configure your own certificate.
```

[id="{p}-tls-certificates"]
== TLS certificates

> This section only covers TLS certificates for the HTTP layer. The TLS certificates for the transport layer used for Elasticsearch internal communication between Elasticsearch nodes in a cluster are managed by ECK and are not configurable.

Reviewer: ```suggestion
This section only covers TLS certificates for the HTTP layer. TLS certificates for the transport layer -- used for internal communications between Elasticsearch nodes -- are managed by ECK and cannot be changed.
```

}
----

> As an alternative, you can edit the spec of the Elasticsearch resource to include the external IP in the `subjectAltNames` element (SANs), as in the example below. This is useful for clients where `--resolve` information cannot be specified, or when it is more convenient to make changes on the server rather than the client.

Reviewer: ```suggestion
You can add extra Subject Alternative Name (SAN)  entries to the generated certificate by setting the `subjectAltNames` field of the Elasticsearch specification as illustrated in the example below. Adding the external IP address to the SANs eliminates the need to explicitly configure the clients to trust the certificate exposed by the server.
```
